content/en/docs/concepts/workloads/pods/_index.md

```diff
@@ -191,6 +191,30 @@ details are abstracted away. That abstraction and separation of concerns simplif
 system semantics, and makes it feasible to extend the cluster's behavior without
 changing existing code.
 
+## Pod update and replacement
+
+As mentioned in the previous section, when the Pod template for a workload
+resource is changed, the controller will create new Pods based on the updated
+template instead of updating or patching the existing Pods.
+
+Kubernetes doesn't prevent you from managing Pods directly. It is possible to
+update a Pod in place. However, the Pod update operation has some limitations:
+
+- Most of the metadata about a Pod is immutable. For example, you cannot
+  change the `namespace`, `name`, `uid`, `creationTimestamp` fields, you can
+  only increment the `generation` field if you want to change it.
+- If the `metadata.deletionTimestamp` is set, no new entry can be added to the
+  `metadata.finalizers` list.
```

Comment:
Maybe mention that this is not specific to Pods (but! I can't think of some concise wording).

Reply:
Maybe this is already done in the GC concept/task.